orchestration/dagster_orchestration/hca_manage/dataset.py

```diff
@@ -62,8 +64,15 @@ def _remove_dataset(args: argparse.Namespace) -> None:
     hca.delete_dataset(dataset_name=args.dataset_name, dataset_id=args.dataset_id)
 
 
+# validate provided name against dataset name regex from the spec
+def validate_dataset_name(dataset_name: str) -> None:
```

Comment:
⛏️ Prefix this with a `_` to indicate it's an internal method that clients should not be calling.